emspy/query/profile.py

```diff
@@ -66,24 +102,77 @@ def get_profile_results(self, flight_id):
+        if self._search_type == 'name':
+            if self._exact_search is True:  # name + exact search
+                # locate profiles using exact string match.
+                filtered = res.loc[res['name'] == self._input_profile_name, :]
+            elif self._exact_search is False:  # name + inexact search
+                # locate profiles with string in name (case insensitive).
+                res['name_lower'] = res['name'].str.lower()
+                filtered = res.loc[res['name_lower'].str.contains(self._input_profile_name.lower()), :]
+        elif self._search_type == 'number':  # number search
+            # locate rows where the localId matches the input profile number.
+            filtered = res.loc[res['localId'] == self._input_profile_number]
+        return filtered
+
+    def __validate_filtered(self, filtered):
+        # Check the filtered values.
+        # If only one row is found, set the class attributes using this row.
         if len(filtered) == 1:
             print('Found a profile with the supplied profile number and name.')
-            self._guid = filtered['id'].values[0]
-            self._current_version = filtered['currentVersion'].values[0]
-            self._library = filtered['library'].values[0]
-            self._profile_name = filtered['name'].values[0]
-            self._local_id = filtered['localId'].values[0]
-            print('Profile name: {0}, profile number: {1}.'.format(self._profile_name, self._local_id))
-        elif len(filtered < 0):
-            print(
-                'No profile found with the supplied profile number and name.  Perhaps try without the name (it could'
-                ' take longer).')
-        elif len(filtered > 1):
-            print('Somehow found multiple profiles with the supplied name and number.')
+            print('Profile name: {0}, profile number: {1}.'.format(filtered['localId'].iloc[0],
+                                                                   filtered['name'].iloc[0]))
+            self.__set_profile_attributes(filtered)
+        # if more than one rows are found (shouldn't be possible) either try again (if we are searching by name in an
+        # inexact manner, we will now try an exact search) or return a LookupError.
+        # Inexact search could learn to more than one match, leaving the potential for an exact search to narrow to one.
+        elif len(filtered) > 1:
+            match_names = filtered['name'].values
+            match_profiles = filtered['localId'].values
+            match_strings = ['P' + str(pnum) + ': ' + name for pnum, name in zip(match_profiles, match_names)]
+            match_names_formatted = '\t' + '\n\t'.join(match_strings)  # format a string for printing
+            print(self._many_matches_name_template.format(self._input_profile_name, match_names_formatted))
+            if self._search_type == 'name' and self._exact_search is False:
```

Comment:
Might want to get rid of the `_search_type` check since we can only get here for a name search